CMakeExternals/ITK_seach_gpu.patch

```diff
@@ -0,0 +1,354 @@
+   cl_int errid;
+-  for(unsigned int i=0; i<m_NumberOfDevices; i++)
++  for(unsigned int i=0; i<m_DevicesList.size(); i++)
+     {
+     errid = clReleaseCommandQueue(m_CommandQueue[i]);
+     OpenCLCheckError( errid, __FILE__, __LINE__, ITK_LOCATION );
+@@ -88,33 +96,33 @@ GPUContextManager::~GPUContextManager()
+   free(m_CommandQueue);
+   errid = clReleaseContext(m_Context);
+   OpenCLCheckError( errid, __FILE__, __LINE__, ITK_LOCATION );
+-  if(m_NumberOfDevices > 0)
++  
++  if (m_DevicesList.size())
+     {
+-    free(m_Devices);
++      m_DevicesList.clear();
+     }
+ }
+ 
+ cl_command_queue GPUContextManager::GetCommandQueue(int i)
+ {
+-  if(i < 0 || i >= (int)m_NumberOfDevices)
++  if(i < 0 || i >= (int)m_DevicesList.size())
+     {
+     printf("Error: requested queue id is not available. Default queue will be used (queue id = 0)\n");
+     return m_CommandQueue[0];
+     }
+ 
+-//std::cout << "Command queue " << i << " is requested " << std::endl;
+-
+   return m_CommandQueue[i];
+ }
+ 
+ cl_device_id GPUContextManager::GetDeviceId(int i)
+ {
+-  if(i < 0 || i >= (int)m_NumberOfDevices)
++  if(i < 0 || i >= (int)m_DevicesList.size())
+     {
+     printf("Error: requested queue id is not available. Default queue will be used (queue id = 0)\n");
+-    return m_Devices[0];
++  return m_DevicesList[0];
+     }
+-  return m_Devices[i];
++
++  return m_DevicesList[i];
+ }
+ 
+ } // namespace itk
+diff --git a/Modules/Core/GPUCommon/src/itkOpenCLUtil.cxx b/Modules/Core/GPUCommon/src/itkOpenCLUtil.cxx
+index 21ebddb..65a5039 100644
+--- a/Modules/Core/GPUCommon/src/itkOpenCLUtil.cxx
++++ b/Modules/Core/GPUCommon/src/itkOpenCLUtil.cxx
+@@ -185,6 +185,50 @@ void OpenCLPrintDeviceInfo(cl_device_id device, bool verbose)
+   }
+ }
+ 
++std::vector<cl_platform_id> OpenCLGetPlatformsList()
++{
++  cl_uint num_platforms;
++  cl_int ciErrNum;
++  std::vector<cl_platform_id> platformsList;
++
++  ciErrNum = clGetPlatformIDs(0, ITK_NULLPTR, &num_platforms);
++  if (ciErrNum != CL_SUCCESS)
++    {
++    printf("clGetPlatformIDs: get number of platforms error. Error code = %i \n\n", ciErrNum);
++    return std::vector<cl_platform_id>();
++    }
++  
++  if(num_platforms == 0)
++    {
++    printf("No OpenCL platform found!\n\n");
++    return std::vector<cl_platform_id>();
++    }
++  
++  platformsList.resize(num_platforms);
++  ciErrNum = clGetPlatformIDs(num_platforms, &platformsList[0], ITK_NULLPTR);
++  if (ciErrNum != CL_SUCCESS)
++    {
++    printf("clGetPlatformIDs: get platforms list error. Error code = %i \n\n", ciErrNum);
++    return std::vector<cl_platform_id>();
++    }
++  
++  std::vector<cl_platform_id>::iterator iter = platformsList.begin();
++  for (; iter != platformsList.end(); ++iter)
++    {
++    std::string platformName;
++    unsigned int platformNameSize = 1024;
++    platformName.resize(platformNameSize);
++    ciErrNum = clGetPlatformInfo(*iter, CL_PLATFORM_NAME, platformName.size(), &platformName[0], &platformNameSize);
```

Comment:
/root/contrib/W/SB/MITK-superbuild/ep/src/ITK/Modules/Core/GPUCommon/src/itkOpenCLUtil.cxx: In function ‘std::vector<_cl_platform_id*> itk::OpenCLGetPlatformsList()’:
/root/contrib/W/SB/MITK-superbuild/ep/src/ITK/Modules/Core/GPUCommon/src/itkOpenCLUtil.cxx:221:115: error: cannot convert ‘unsigned int*’ to ‘std::size_t* {aka long unsigned int*}’ for argument ‘5’ to ‘cl_int clGetPlatformInfo(cl_platform_id, cl_platform_info, std::size_t, void*, std::size_t*)’
     ciErrNum = clGetPlatformInfo(*iter, CL_PLATFORM_NAME, platformName.size(), &platformName[0], &platformNameSize);
                                                                                                                   ^